backends/dpdk/dpdkProgram.cpp

```diff
@@ -187,6 +187,75 @@ IR::Declaration_Variable *ConvertToDpdkParser::addNewTmpVarToMetadata(cstring na
     return newTmpVar;
 }
 
+/* This is a helper function for handling the transition select statement. It populates the left
+   and right operands of comparison, to be used  in conditional jump instructions. When the keyset
+   of select case is simple expressions, it populates the left and rigt operands with the input
+   expressions. When the keyset is Mask expression "a &&& b", it inserts temporary variables in
+   Metadata structure and populates the left and right operands of comparison with
+   "input & b" and "a & b" */
+void ConvertToDpdkParser::getCondVars(const IR::Expression *sv, const IR::Expression *ce,
+                                      IR::Expression **leftExpr, IR::Expression **rightExpr) {
+    if (auto maskexpr = ce->to<IR::Mask>()) {
+        auto left = maskexpr->left;
+        auto right = maskexpr->right;
+        /* Dpdk architecture requires that both operands of &&& be constants */
+        if (!left->is<IR::Constant>() || !left->is<IR::Constant>()) {
```

Comment:
You are using the same expression twice.

Reply:
Thanks for catching this. When I tried hitting this error with a testcase, I see this condition is already checked by midend in DoSimplifySelectCases pass. So will remove this error check altogether.